instrumentation/akka/akka-http-10.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/akkahttp/server/route/AkkaRouteHolder.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.akkahttp.server.route;
+
+import static io.opentelemetry.context.ContextKey.named;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import io.opentelemetry.context.ImplicitContextKeyed;
+import io.opentelemetry.instrumentation.api.semconv.http.HttpServerRoute;
+import io.opentelemetry.instrumentation.api.semconv.http.HttpServerRouteSource;
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+public class AkkaRouteHolder implements ImplicitContextKeyed {
+  private static final ContextKey<AkkaRouteHolder> KEY = named("opentelemetry-akka-route");
+
+  private String route = "";
+  private boolean newSegment;
+  private boolean endMatched;
+  private final Deque<String> stack = new ArrayDeque<>();
+
+  public static Context init(Context context) {
+    if (context.get(KEY) != null) {
+      return context;
+    }
+    return context.with(new AkkaRouteHolder());
+  }
+
+  public static void push(String path) {
+    AkkaRouteHolder holder = Context.current().get(KEY);
+    if (holder != null && holder.newSegment && !holder.endMatched) {
+      holder.route += path;
```

Comment:
This assumes or has knowledge that the `path` has delimiters baked into it? Like a leading slash or something?

Reply:
The path that is passed here can be `/`, so yes, delimiters are already in the path.